File: MODUL6/PraktikumC/PRAK605-2310817110004-ABDURRAHMANGILANGHARJUNA.c
#include <stdio.h>

void perkalianMatriks(int n, int A[n][n], int B[n][n], int hasil[n][n]) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            hasil[i][j] = 0;
            for (int k = 0; k < n; k++) {
                hasil[i][j] += A[i][k] * B[k][j];
            }
        }
    }
}

void tampilkanMatriks(int n, int matrix[n][n]) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int n;
    scanf("%d", &n);

    int A[n][n], B[n][n], hasil[n][n];
    printf("Matriks A:\n");
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            scanf("%d", &A[i][j]);
        }
    }

    printf("Matriks B:\n");
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            scanf("%d", &B[i][j]);
        }
    }

    perkalianMatriks(n, A, B, hasil);
    printf("Matriks AXB:\n");
    tampilkanMatriks(n, hasil);

    return 0;
}

File: MODUL6/PraktikumPy/PRAK605-2310817110004-ABDURRAHMANGILANGHARJUNA.py
def perkalian_matriks(A, B):
    n = len(A)
    hasil = [[0] * n for _ in range(n)]

    for i in range(n):
        for j in range(n):
            for k in range(n):
                hasil[i][j] += A[i][k] * B[k][j]

    return hasil

def main():
    n = int(input())
    print("Matriks A")
    A = [list(map(int, input().split())) for _ in range(n)]
    print("Matriks B")
    B = [list(map(int, input().split())) for _ in range(n)]

    hasil = perkalian_matriks(A, B)
    print("Matriks AXB")
    for row in hasil:
        print(*row)

if __name__ == "__main__":
    main()
